x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -488,135 +416,151 @@ export const errorAggregator = (
  * @param ruleParamsMaxSignals int representing the maxSignals property on the rule (usually unmodified at 100)
- * @param gap moment.Duration representing a gap in since the last time the rule ran
- * @param previousStartedAt Date at which the rule last ran
- * @param interval string the interval which the rule runs
  * @param buildRuleMessage function provides meta information for logged event
  */
-export const getSignalTimeTuples = ({
-  logger,
+export const getOriginalTuple = ({
   ruleParamsFrom,
   ruleParamsTo,
   ruleParamsMaxSignals,
-  gap,
-  previousStartedAt,
-  interval,
   buildRuleMessage,
 }: {
-  logger: Logger;
   ruleParamsFrom: string;
   ruleParamsTo: string;
   ruleParamsMaxSignals: number;
-  gap: moment.Duration | null;
-  previousStartedAt: Date | null | undefined;
-  interval: string;
   buildRuleMessage: BuildRuleMessage;
-}): Array<{
-  to: moment.Moment | undefined;
-  from: moment.Moment | undefined;
-  maxSignals: number;
-}> => {
-  let totalToFromTuples: Array<{
-    to: moment.Moment | undefined;
-    from: moment.Moment | undefined;
-    maxSignals: number;
-  }> = [];
-  if (gap != null && gap.valueOf() > 0 && previousStartedAt != null) {
-    const fromUnit = ruleParamsFrom[ruleParamsFrom.length - 1];
-    if (isValidUnit(fromUnit)) {
-      const unit = fromUnit; // only seconds (s), minutes (m) or hours (h)
-
-      /*
-      we need the total duration from now until the last time the rule ran.
-      the next few lines can be summed up as calculating
-      "how many second | minutes | hours have passed since the last time this ran?"
-      */
-      const nowToGapDiff = moment.duration(moment().diff(previousStartedAt));
-      const calculatedFrom = `now-${
-        parseInt(shorthandMap[unit].asFn(nowToGapDiff).toString(), 10) + unit
-      }`;
-      logger.debug(buildRuleMessage(`calculatedFrom: ${calculatedFrom}`));
-
-      const intervalMoment = moment.duration(parseInt(interval, 10), unit);
-      logger.debug(buildRuleMessage(`intervalMoment: ${shorthandMap[unit].asFn(intervalMoment)}`));
-      const momentUnit = shorthandMap[unit].momentString as moment.DurationInputArg2;
-      // maxCatchup is to ensure we are not trying to catch up too far back.
-      // This allows for a maximum of 4 consecutive rule execution misses
-      // to be included in the number of signals generated.
-      const { maxCatchup, ratio, gapDiffInUnits } = getGapMaxCatchupRatio({
-        logger,
-        buildRuleMessage,
-        previousStartedAt,
-        unit,
-        ruleParamsFrom,
-        interval,
-      });
-      logger.debug(buildRuleMessage(`maxCatchup: ${maxCatchup}, ratio: ${ratio}`));
-      if (maxCatchup == null || ratio == null || gapDiffInUnits == null) {
-        throw new Error(
-          buildRuleMessage('failed to calculate maxCatchup, ratio, or gapDiffInUnits')
-        );
-      }
-      let tempTo = dateMath.parse(ruleParamsFrom);
```

Comment:
tempTo is converted from a "now" based timestamp to a concrete moment here. However, time keeps advancing before ruleParamsFrom is converted into a concrete timestamp for the current rule run tuple at the end. Thus we can end up with a small gap between some of the tuples.